EventFilter/GEMRawToDigi/plugins/GEMDigiToRawModule.cc

```diff
@@ -123,37 +123,36 @@ void GEMDigiToRawModule::produce(edm::StreamID iID, edm::Event& iEvent, edm::Eve
     }
   }
 
-  for (unsigned int fedId = FEDNumbering::MINGEMFEDID; fedId <= FEDNumbering::MAXGEMFEDID; ++fedId) {
+  for (unsigned int fedId = FEDNumbering::MINGEMFEDID; fedId <= FEDNumbering::MAXME0FEDID; ++fedId) {
     uint32_t amc13EvtLength = 0;
     std::unique_ptr<AMC13Event> amc13Event = std::make_unique<AMC13Event>();
 
-    for (auto const& gemBx : gemBxMap) {
-      int bx = gemBx.first;
-      GEMDigiCollection inBxGemDigis = gemBx.second;
+    for (uint8_t amcNum = 0; amcNum < GEMeMap::maxAMCs_; ++amcNum) {
+      uint32_t amcSize = 0;
+      std::unique_ptr<AMCdata> amcData = std::make_unique<AMCdata>();
 
-      for (uint8_t amcNum = 0; amcNum < GEMeMap::maxAMCs_; ++amcNum) {
-        uint32_t amcSize = 0;
-        std::unique_ptr<AMCdata> amcData = std::make_unique<AMCdata>();
+      for (uint8_t gebId = 0; gebId < GEMeMap::maxGEBs_; ++gebId) {
+        std::unique_ptr<GEBdata> gebData = std::make_unique<GEBdata>();
+        GEMROMapping::chamEC geb_ec{fedId, amcNum, gebId};
 
-        for (uint8_t gebId = 0; gebId < GEMeMap::maxGEBs_; ++gebId) {
-          std::unique_ptr<GEBdata> gebData = std::make_unique<GEBdata>();
-          GEMROMapping::chamEC geb_ec{fedId, amcNum, gebId};
+        if (!gemROMap->isValidChamber(geb_ec))
+          continue;
+        GEMROMapping::chamDC geb_dc = gemROMap->chamberPos(geb_ec);
 
-          if (!gemROMap->isValidChamber(geb_ec))
-            continue;
-          GEMROMapping::chamDC geb_dc = gemROMap->chamberPos(geb_ec);
+        auto vfats = gemROMap->getVfats(geb_dc.detId);
+        for (auto const& vfat_ec : vfats) {
+          GEMROMapping::vfatDC vfat_dc = gemROMap->vfatPos(vfat_ec);
+          GEMDetId gemId = vfat_dc.detId;
+          uint16_t vfatId = vfat_ec.vfatAdd;
 
-          auto vfats = gemROMap->getVfats(geb_dc.detId);
-          for (auto const& vfat_ec : vfats) {
-            GEMROMapping::vfatDC vfat_dc = gemROMap->vfatPos(vfat_ec);
-            GEMDetId gemId = vfat_dc.detId;
-            uint16_t vfatId = vfat_ec.vfatAdd;
+          for (auto const& gemBx : gemBxMap) {
+            int bc = amcData->bx() + gemBx.first;
```

Comment:
actually that header we talked about in https://github.com/cms-sw/cmssw/pull/31863#discussion_r509607308 is effectively used here. So, instead of 255 (uint8_t(-1)) we are getting 0 while packing. But then during unpacking the AMC BX is 255, which leads to the bad shift observed in the tests.

Reply:
this is fix. it should have been 
```
int bc = BX_id + gemBx.first;
```